### PR TITLE
raidboss: Add Option to Order Timeline Bottom-to-Top

### PR DESCRIPTION
**Pull request by quisquous**
This is #4811 with conflicts resolved due to refactoring and 3ae3f6c1eca6dd6a96d515d7d54ed654f5e9642e on top. 3ae3f6c1eca6dd6a96d515d7d54ed654f5e9642e moves the keep alive logic from html_timeline_ui to timeline so that it knows about how many bars are still on screen and also knows when a bar has been expired and it should consider adding more to the list.

Closes #4811.

### CODE REVIEW

File: ui/raidboss/timeline.ts [hunk 1]
```diff
@@ -342,16 +352,45 @@ export class Timeline {
...
+            () => {
+              // Find and remove the first keepalive event with this id.
+              let found = false;
+              this.keepAliveEvents = this.keepAliveEvents.filter((x) => {
+                if (found)
+                  return true;
+                if (x.event.id === event.id) {
+                  found = true;
+                  return false;
+                }
```
Comment (panicstevenson) [inline ui/raidboss/timeline.ts:380]: Does this need to `return true;` in the fallthrough where it is not found and the ids are not equal?

File: ui/raidboss/timeline.ts [hunk 2]
```diff
@@ -342,16 +352,45 @@ export class Timeline {
...
 
     this.activeEvents = durationEvents;
   }
 
   private _RemoveExpiredTimers(fightNow: number): void {
     let activeEvent = this.activeEvents[0];
     while (this.activeEvents.length && activeEvent && activeEvent.time <= fightNow) {
-      this.ui?.OnRemoveTimer(activeEvent, true);
+      const event = activeEvent;
+      if (this.options.KeepExpiredTimerBarsForSeconds > 0) {
```
Comment (panicstevenson) [inline ui/raidboss/timeline.ts:367]: (This introduces regression in the user experience where a previously valid negative KeepExpiredTimerBarsForSeconds value was set to the HTML5 minimum of [4ms](https://developer.mozilla.org/en-US/docs/Web/API/setTimeout#Specification). /s)

**Review by panicstevenson — APPROVED** [review]
Nice! Thank you for bringing this across the finish line.

File: ui/raidboss/timeline.ts (hunk 1 above)
Comment (quisquous) [reply ui/raidboss/timeline.ts:380]: Oh yeah, thanks!

**Pull request merged**